modules/apps/journal/journal-content-web/src/main/resources/META-INF/resources/js/printPageButtonPropsTransformer.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {printPage} from 'frontend-js-web';
+
+export default function ({additionalProps, portletNamespace}) {
+	const printPageButton = document.getElementById(
+		`${portletNamespace}printPageButton`
+	);
+
+	if (printPageButton) {
+		printPageButton.addEventListener('click', () => {
+			printPage(additionalProps.printPageURL);
+		});
+	}
```

Comment:
Mentioned above, but a `propsTransformer` should be returning an object of new props. So you can return `{onClick: //... }`